homeassistant/components/sense/sensor.py

```diff
@@ -46,64 +54,172 @@ def __init__(self, name, sensor_type):
 SENSOR_VARIANTS = [PRODUCTION_NAME.lower(), CONSUMPTION_NAME.lower()]
 
 
+def sense_to_mdi(sense_icon):
+    """Convert sense icon to mdi icon."""
+    return "mdi:{}".format(MDI_ICONS.get(sense_icon, "power-plug"))
+
+
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Sense sensor."""
     data = hass.data[DOMAIN][config_entry.entry_id][SENSE_DATA]
+    sense_devices_data = hass.data[DOMAIN][config_entry.entry_id][SENSE_DEVICES_DATA]
 
     @Throttle(MIN_TIME_BETWEEN_DAILY_UPDATES)
     async def update_trends():
         """Update the daily power usage."""
         await data.update_trend_data()
 
-    async def update_active():
-        """Update the active power usage."""
-        await data.update_realtime()
-
     sense_monitor_id = data.sense_monitor_id
+    sense_devices = hass.data[DOMAIN][config_entry.entry_id][
+        SENSE_DISCOVERED_DEVICES_DATA
+    ]
+    await data.update_trend_data()
+
+    devices = [
+        SenseEnergyDevice(sense_devices_data, device, sense_monitor_id)
+        for device in sense_devices
+        if device["tags"]["DeviceListAllowed"] == "true"
+    ]
+
+    for var in SENSOR_VARIANTS:
+        name = ACTIVE_SENSOR_TYPE.name
+        sensor_type = ACTIVE_SENSOR_TYPE.sensor_type
+        is_production = var == PRODUCTION_NAME.lower()
```

Comment:
why can't we just make `PRODUCTION_NAME` be the correct casing ?

Reply:
Its used building the name

```
        name_type = PRODUCTION_NAME if is_production else CONSUMPTION_NAME
        self._name = f"{name} {name_type}"
```

Reply:
I'll make them constants so we can avoid all the lower()s

```
CONSUMPTION_NAME = "Usage"
CONSUMPTION_ID = "usage"
PRODUCTION_NAME = "Production"
PRODUCTION_ID = "production"
```